File: B-No.5073.py
# src: https://www.acmicpc.net/problem/5073
while True:
    l=input()
    if l=="0 0 0":
        break
    a,b,c=map(int,l.split())
    if max(a,b,c) >= a+b+c-max(a,b,c):
        print("Invalid")
    elif a==b==c:
        print("Equilateral")
    elif a==b or b==c or a==c:
        print("Isosceles")
    else:
        print("Scalene")    

File: B-No.5073.cpp
// src : https://www.acmicpc.net/problem/5073
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    int a[3];
    while(true)
    {
        for(int i=0;i<3;i++)
            cin >> a[i];
        sort(a,a+3);

        if(a[2]==0)
            break;
        else if(a[0]+a[1]<=a[2])
            cout<<"Invalid"<<endl;
        else if(a[0]==a[2])   
            cout<<"Equilateral"<<endl;
        else if(a[1]==a[2] || a[0]==a[1])
            cout<<"Isosceles"<<endl;
        else
            cout<<"Scalene"<<endl; 
    }
    return 0;
}